File: src/main/java/com/heekng/celloct/controller/JoinRequestController.java
package com.heekng.celloct.controller;

import com.heekng.celloct.config.oauth.dto.SessionMember;
import com.heekng.celloct.dto.JoinRequestDto;
import com.heekng.celloct.service.JoinRequestService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;

import javax.servlet.http.HttpSession;

@Controller
@Slf4j
@RequiredArgsConstructor
@RequestMapping("/joinRequest")
public class JoinRequestController {

    private final HttpSession httpSession;
    private final JoinRequestService joinRequestService;

    @PostMapping
    @ResponseBody
    public Boolean joinRequest(@RequestBody JoinRequestDto.JoinRequest joinRequest) {
        SessionMember sessionMember = (SessionMember) httpSession.getAttribute("member");
        joinRequest.setMemberId(sessionMember.getId());

        try {
            Long joinRequestId = joinRequestService.joinRequest(joinRequest);
        } catch (Exception e) {
            return false;
        }
        return true;
    }
}

File: src/main/kotlin/com/heekng/celloct/controller/JoinRequestController.kt
package com.heekng.celloct.controller

import com.heekng.celloct.config.oauth.dto.SessionMember
import com.heekng.celloct.dto.JoinRequestDto
import com.heekng.celloct.service.JoinRequestService
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseBody
import javax.servlet.http.HttpSession

@Controller
@RequestMapping("/joinRequest")
class JoinRequestController(
    private val httpSession: HttpSession,
    private val joinRequestService: JoinRequestService,
) {

    @PostMapping
    @ResponseBody
    fun joinRequest(
        @RequestBody joinRequest: JoinRequestDto.JoinRequest,
    ): Boolean {
        val sessionMember = httpSession.getAttribute("member") as SessionMember?
        joinRequest.memberId = sessionMember!!.id

        try {
            val joinRequestId = joinRequestService.joinRequest(joinRequest)
        } catch (e: Exception) {
            return false
        }
        return true
    }
}